src/ledger/LedgerTxnTrustLineSQL.cpp

```diff
@@ -88,25 +100,11 @@ LedgerTxnRoot::Impl::insertOrUpdateTrustLine(LedgerEntry const& entry,
 {
     auto const& tl = entry.data.trustLine();
 
-    std::string actIDStrKey = KeyUtils::toStrKey(tl.accountID);
-    unsigned int assetType = tl.asset.type();
-    std::string issuerStr, assetCode;
-    if (tl.asset.type() == ASSET_TYPE_CREDIT_ALPHANUM4)
-    {
-        issuerStr = KeyUtils::toStrKey(tl.asset.alphaNum4().issuer);
-        assetCodeToStr(tl.asset.alphaNum4().assetCode, assetCode);
-    }
-    else if (tl.asset.type() == ASSET_TYPE_CREDIT_ALPHANUM12)
-    {
-        issuerStr = KeyUtils::toStrKey(tl.asset.alphaNum12().issuer);
-        assetCodeToStr(tl.asset.alphaNum12().assetCode, assetCode);
-    }
-    if (actIDStrKey == issuerStr)
-    {
-        throw std::runtime_error("Issuer's own trustline should not be used "
-                                 "outside of OperationFrame");
-    }
+    std::string accountIDStr, issuerStr, assetCodeStr;
+    getTrustLineStrings(tl.accountID, tl.asset, accountIDStr, issuerStr,
```

Comment:
all of those looks like unrelated change, would you mind extracting to separate commit?

Reply:
fixed.